provider/resource_keycloak_openid_client_service_account_role.go

```diff
@@ -40,18 +40,32 @@ func resourceKeycloakOpenidClientServiceAccountRole() *schema.Resource {
 	}
 }
 
-func getOpenidClientServiceAccountRoleFromData(data *schema.ResourceData) *keycloak.OpenidClientServiceAccountRole {
-	return &keycloak.OpenidClientServiceAccountRole{
-		Id:                   data.Id(),
-		ContainerId:          data.Get("client_id").(string),
-		Name:                 data.Get("role").(string),
-		RealmId:              data.Get("realm_id").(string),
-		ServiceAccountUserId: data.Get("service_account_user_id").(string),
+func getOpenidClientServiceAccountRoleFromData(data *schema.ResourceData, keycloakClient *keycloak.KeycloakClient) (*keycloak.OpenidClientServiceAccountRole, error) {
+
+	containerId := data.Get("client_id").(string)
+	roleName := data.Get("role").(string)
+	realmId := data.Get("realm_id").(string)
+	serviceAccountRoleId := data.Get("service_account_user_id").(string)
+
+	if realmId != "" && containerId != "" {
```

Comment:
`realm_id` and `client_id` are both required attributes, so I think this statement is always `true`.

Reply:
They can be empty if you're using ```realm_id = keycloak_realm.realm.id ``` and ```client_id  = keycloak_openid_client.client.id``` that do not exist yet